packages/data-service/src/connection-info.ts

```diff
@@ -33,3 +33,6 @@ interface ConnectionFavoriteOptions {
    */
   color?: string;
 }
+
+export type ConnectionInfoWithRequiredId = ConnectionInfo &
```

Comment:
I was noticing that we made `id` as readOnly in `ConnectionInfo`, so probably we could just make id required by default in `ConnectionInfo`, not sure if it would require to change too many tests, but probably makes sense. Not that important though.

Reply:
Made id required 👍